pages/onboarding/index.tsx

```diff
@@ -1,15 +1,131 @@
+import React, { useState, useEffect } from 'react';
+import Cookies from 'universal-cookie';
+
+import createAxiosInstance from 'client/lib/axios';
+import { UserProfile } from 'server/types/User';
+import { AUTH_COOKIE_NAME } from 'shared/constants';
+
+interface FormData {
+  bio: string;
+  twitter_username: string;
+  linkedin_url: string;
+  github_username: string;
+  website: string;
+}
+
+interface Props {
+  user: UserProfile | null;
+}
+
+const OnboardingPage: NextPage<Props> = ({ user }: Props) => {
+  const [formData, setFormData] = useState<FormData>({
+    bio: user?.bio,
+    twitter_username: user?.twitter_username,
+    linkedin_url: user?.linkedin_url,
+    github_username: user?.github_username,
+    website: user?.website,
+  });
+
+  const [isLoading, setIsLoading] = useState(false);
+  const [isFormFilled, setIsFormFilled] = useState(false);
+
+  const router = useRouter();
+
+  useEffect(() => {
+    if (!user) {
+      router.push('/');
+    }
+  }, [user]);
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { name, value } = event.target;
+    setFormData(prevFormData => ({ ...prevFormData, [name]: value }));
+    const isFormFilled = Object.values(formData).some(value => !!value);
+    setIsFormFilled(isFormFilled);
+  };
+
+  const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    try {
+      setIsLoading(true);
+      await axios.patch(`/api/users/${user?.id}`, formData);
+      setIsLoading(false);
+      router.push('/directory');
+    } catch (error) {
+      setIsLoading(false);
+      alert(error);
+    }
+  };
+
+  const isSubmitDisabled = !isFormFilled && !isLoading;
 
-const Onboarding: NextPage = () => {
   return (
     <Container maxWidth="lg" className="flex justify-between pt-4">
       <Typography variant="h1" className="text-3xl">
         Onboarding
       </Typography>
+      <form onSubmit={handleSubmit}>
+        <TextField
+          label="Bio"
+          name="bio"
+          placeholder='Describe yourself!'
+          value={formData.bio}
+          onChange={handleChange}
+          multiline
+          rows={4}
+          fullWidth
+          inputProps={{ maxLength: 1000 }}
+        />
+        <TextField
+          label="Twitter Username"
+          placeholder='JonDoe24'
+          name="twitter_username"
+          value={formData.twitter_username}
+          onChange={handleChange}
+          fullWidth
+          inputProps={{ maxLength: 200 }}
+        />
+        <TextField
+          label="LinkedIn URL"
+          name="linkedin_url"
+          placeholder='https://www.linkedin.com/in/your-profile'
+          value={formData.linkedin_url}
+          onChange={handleChange}
+          fullWidth
+          inputProps={{ maxLength: 200 }}
+        />
+        <TextField
+          label="GitHub Username"
+          name="github_username"
+          placeholder='Jon Doe'
```

Comment:
this technically isn't a valid username, also i think we should avoid gendered examples - would change this (plus the placeholder twitter handle above) to be more generic, e.g. `twitteruser` and `githubuser`